drs3/api/main.py

```diff
@@ -96,7 +99,63 @@ def get_objects_id(
     Returns:
         An instance of ``DrsReturnObject``
     """
-    ...
+
+    object_id = request.matchdict["object_id"]
+    config: Config = CONFIG
+
+    with Database(config=config) as database:
+        try:
+            db_object_info = database.get_drs_object(object_id)
+        except DrsObjectNotFoundError as object_not_found_error:
+            raise HTTPNotFound(
+                json={
+                    "msg": "The requested DRSObject does not exist",
+                    "status_code": 404,
+                }
+            ) from object_not_found_error
+
+    # If object exists in Database, see if it exists in outbox
+
+    bucket_id = config.s3_outbox_bucket_id
+
+    with ObjectStorage(config=config) as storage:
+        if storage.does_object_exist(bucket_id, object_id):
+
+            # create presigned url
+            response = storage.get_object_download_url(bucket_id, object_id)
+
+            # change path to localhost
+            path = "http://localhost:4566" + response.removeprefix(
+                config.s3_endpoint_url
+            )
+
+            return DrsObjectServe(
+                id=object_id,
+                self_uri=f"{config.drs_self_url}/{object_id}",
+                size=db_object_info.size,
+                created_time=db_object_info.registration_date,
+                checksums=[Checksum(checksum=db_object_info.md5_checksum, type="md5")],
+                access_methods=[
+                    AccessMethod(access_url=AccessURL(url=path), type="s3")
+                ],
+            )
+
+        # Publish non_staged_file_requested
+        message = {
+            "request_id": None,
+            "file_id": db_object_info.id,
+            "drs_id": db_object_info.external_id,
+            "timestamp": db_object_info.registration_date,
+        }
+        publish_topic(
+            config.rabbitmq_host,
+            config.rabbitmq_port,
+            "non_staged_file_requested",
+            message,
+        )
```

Comment:
Could we put this into the core subpackage as a separate function? We may also consider to remove the pyramid exception (HTTPNotFound) and just let it throw the DrsObjectNotFoundError. We can still "mention" it in the core function just to make sure everybody reading this code understands what can happen, e.g. like this:
```
# check if DRS object is registered in the database
# and forward the error if not:
try:
    db_object_info = database.get_drs_object(object_id)
except DrsObjectNotFoundError:
    raise
```

The DrsObjectNotFoundError can then be translated to the HTTP error in the view function as you did here.


The last part of publishing a message should however not be part of the core function directly. One possibility would be to give the core function and argument of type Callable called something like "make_stage_request". Then when calling the core function in the view function we hand over the appropriate publishing function. The publishing function then receives a pydantic model of the DRS object and constructs and sends the message.

Like this we would achieve a nice separation between business logic and the API.